[Validate] Fix logging error (#3346)
demisto/demisto-sdk: CHANGELOG.md

* Fixed an issue where **validate** would not properly detect dependencies of core packs.
* Removed usages of Random in unit tests to ensure the tests are deterministic.
* **validate** will now run on all the pack content items when the pack supported marketplaces are modified.
* **pre-commit** no longer runs when there are no modified files (unless provided with input files).
* Fixed an issue where errors in **validate** were logged as `info`.

> * Fixed an issue where **validate** error messages were not logged when the `param` is not valid or if the default field in reputation command is no valid

Reviewer: ```suggestion
* Fixed an issue where **validate** error messages were not logged when an integration param, or the default argument in reputation commands is not valid.
```

* Fixed an issue where **validate** would not properly detect dependencies of core packs.
* Removed usages of Random in unit tests to ensure the tests are deterministic.
* **validate** will now run on all the pack content items when the pack supported marketplaces are modified.

> * **pre-commit** will stop running at the start when there was no change to any files.

Reviewer: ```suggestion
* **pre-commit** no longer runs when there are no modified files (unless provided with input files).
```

* Fixed an issue where errors in **validate** were logged as `info`.

> * Fixed an issue where the error messages were not log in **validate**

Reviewer: ```suggestion
* Fixed an issue where **validate** error messages were not logged
```

> * Fixed an issue where the error messages were not log in **validate**

Reviewer: In what specific cases?
